packages/apps/plugins/plugin-markdown/src/MarkdownPlugin.tsx

```diff
@@ -244,135 +221,49 @@ export const MarkdownPlugin = (): PluginDefinition<MarkdownPluginProvides> => {
-            // TODO(wittjosiah): Autocomplete is not working and this query is causing performance issues.
-            // const query = space?.db.query(Filter.schema(DocumentType));
-            // query?.subscribe();
-            return getCustomExtensions(doc /*, query */);
-          }, [doc, space, settings.values.editorInputMode, getViewMode(doc?.id)]);
-
-          const dispatch = useIntentDispatcher();
-          const handleCommentSelect = useCallback(() => {
-            void dispatch({ action: LayoutAction.SET_LAYOUT, data: { element: 'complementary', state: true } });
-          }, [dispatch]);
-
-          const fileManagerPlugin = useResolvePlugin(parseFileManagerPlugin);
-          const handleFileUpload = useMemo(() => {
-            if (space === undefined) {
-              return undefined;
-            }
-
-            if (fileManagerPlugin?.provides.file.upload === undefined) {
-              return undefined;
-            }
-
-            return async (file: File) => {
-              return fileManagerPlugin?.provides?.file?.upload?.(file, space);
-            };
-          }, [fileManagerPlugin, space]);
-
-          const handleViewModeChange = useCallback(
-            (viewMode: EditorViewMode) => {
-              if (doc) {
-                state.values.viewMode[doc.id] = viewMode;
-              }
-            },
-            [doc?.id],
-          );
 
           switch (role) {
-            // TODO(burdon): Normalize layout (reduce variants).
+            case 'section':
             case 'article': {
-              if (doc) {
+              if (doc && doc.content) {
                 return (
                   <DocumentEditor
-                    viewMode={getViewMode(doc.id)}
-                    toolbar={settings.values.toolbar}
                     document={doc}
-                    extensions={extensions}
-                    onCommentSelect={handleCommentSelect}
-                    onFileUpload={handleFileUpload}
-                    onViewModeChange={handleViewModeChange}
+                    extensionProviders={state.values.extensionProviders}
+                    settings={settings.values}
+                    viewMode={getViewMode(doc.id)}
+                    onViewModeChange={(viewMode: EditorViewMode) => {
+                      if (doc) {
+                        state.values.viewMode[doc.id] = viewMode;
+                      }
+                    }}
+                    role={role}
+                    coordinate={data.coordinate as LayoutCoordinate}
+                    scrollPastEnd
                   />
                 );
-              } else if (
-                data.object &&
-                typeof data.object === 'object' &&
-                'id' in data.object &&
-                typeof data.object.id === 'string' &&
-                'text' in data.object &&
-                typeof data.object.text === 'string'
-              ) {
+              } else if (isEditorModel(data)) {
```

Comment:
```suggestion
              } else if (isEditorModel(data.object)) {
```

Reply:
The guard returns a type at the `data` scope